portal/src/main/webapp/js/src/dashboard/cbio-vendor.js

```diff
@@ -2709,62 +2624,46 @@ window.DataManagerForIviz = (function($, _) {
    */
   content.util.sortClinicalAttrs = function(array) {
     array = array.sort(function(a, b) {
-      var priority = 0;
-      if (content.util.isPriorityClinicalAttr(a.attr_id, a.study_ids)) {
-        if (content.util.isPriorityClinicalAttr(b.attr_id, b.study_ids)) {
-          priority =
-            content.util.compareClinicalAttrsPriority(a.attr_id, b.attr_id);
-        } else {
-          priority = -1;
-        }
-      } else if (content.util.isPriorityClinicalAttr(b.attr_id, b.study_ids)) {
-        priority = 1;
-      } else {
-        priority = 0;
-      }
+      return compareClinicalAttrs(a, b);
+    });
+    return array;
+  };
 
-      if (priority !== 0) {
-        return priority;
-      }
+  function compareClinicalAttrs(a, b) {
+    var priority = 0;
 
-      if (content.util.isPreSelectedClinicalAttr(a.attr_id)) {
-        if (content.util.isPreSelectedClinicalAttr(b.attr_id)) {
-          priority = content.util.compareClinicalAvailability(a, b);
-        }
-      } else if (content.util.isPreSelectedClinicalAttr(b.attr_id)) {
-        priority = 1;
+    if (content.util.isPreSelectedClinicalAttr(a.attr_id)) {
+      if (content.util.isPreSelectedClinicalAttr(b.attr_id)) {
+        priority = content.util.compareClinicalAvailability(a, b);
       } else {
-        priority = 0;
+        priority = -1;
       }
+    } else if (content.util.isPreSelectedClinicalAttr(b.attr_id)) {
+      priority = 1;
+    } else {
+      priority = 0;
+    }
 
-      if (priority !== 0) {
-        return priority;
-      }
+    if (priority !== 0) {
+      return priority;
+    }
 
-      return content.util.compareClinicalAvailability(a, b);
-    });
-    return array;
+    return content.util.compareClinicalAvailability(a, b);
   };
 
   /**
    * Sort clinical attributes by priority.
    * @param {array} array Clinical attributes.
-   * @return {array} Sorted clinical attibutes.
+   * @return {array} Sorted clinical attributes.
    */
   content.util.sortByClinicalPriority = function(array) {
     if (_.isArray(array)) {
       array = array.sort(function(a, b) {
-        var priorityA = a.priority || -1;
-        var priorityB = b.priority || -1;
-
-        if (priorityA === -1) {
-          return 1;
-        }
-        if (priorityB === -1) {
-          return -1;
+        var score = iViz.util.compareLists(a.priority, b.priority, false);
```

Comment:
but it returns a score, right? `var score = iViz.util.compareLists(a.priority, b.priority, false);`